lib/web/terminal.go

```diff
@@ -284,12 +289,43 @@ func (t *TerminalHandler) makeClient(ws *websocket.Conn) (*client.TeleportClient
 	tc.OnShellCreated = func(s *ssh.Session, c *ssh.Client, _ io.ReadWriteCloser) (bool, error) {
 		t.sshSession = s
 		t.windowChange(&t.params.Term)
+
+		go t.startPingLoop(ws)
```

Comment:
why start this here and not in `handler`?

Reply:
I revised to start in handler.

I thought it made more sense to start the loop once session connection is successful b/c if there was an error, then the loop would've been an unnecessary resource allocated only to be torn down right after. But now that I think about it, i see that the loop doesn't have anything to do with sessions.